backend/data/handlers/members.js

```diff
@@ -139,61 +146,99 @@ members.assignTaskToAllMembers = async (filter, newTask) => {
 }
 
 /**
- * Add a new member to the database
+ * Add a new user to the database
  * 
- * @param {Object} memberBody: the new member's details
+ * @param {Object} userPayload:    the new user's details
  */
-members.add = async (memberBody) => {
+members.add = async (userPayload) => {
     return util.handleWrapper(async () => {
-        memberBody = await replaceBodyWithIds(memberBody);
-        return await Member.create(memberBody);
+        // Get fields stored in the Member collection.
+        const userSummaryFields = Object.keys(Member.schema.paths);
+        // Get fields stored in the UserDetails collection.
+        const userDetailFields = Object.keys(UserDetails.schema.paths);
+        // Extract only the fields relevant to the Member collection.
+        let userSummary = _.omit(_.pick(userPayload, userSummaryFields), "_id");
+        // Extract only the fields relevant to the UserDetails collection.
+        let userDetails = _.omit(_.pick(userPayload, userDetailFields), "_id");
+
+        const session = await db.startSession();
+        const response = await session.withTransaction(async () => {
+            const userDetailsResponse = await UserDetails.create(userDetails).session(session);
+            userSummary.miscDetails = userDetailsResponse._id;
+            userSummary = await replacePayloadWithIds(userSummary);
+            const res = await Member.create(userSummary).session(session);
+            return res;
```

Comment:
Everything within the `session.withTransaction` block is a DB transaction (series of DB queries). I will remove the `session.withTransaction` part in Pull Request #2 , since I am having trouble incorporating/testing MongoDB transactions locally.

However, everything **WITHIN** the `session.withTransaction` block will stay. I will just be removing the surrounding `session.withTransaction` part.